From b6b83f91030e3f8e80a6376c7e0d310e4d4dfef9 Mon Sep 17 00:00:00 2001
From: Martin Fick <mfick@codeaurora.org>
Date: Tue, 4 Dec 2012 13:48:19 -0700
Subject: [PATCH] Provide default highlighting of simple active LinkMenuItems

When the Screen displayed matches the Screen which a menu item
would lead to, highlight the active menu (similar to an
active table row).  This highlighting covers most of the
simplest cases but does not handle more complex situations
like the Differences menus, those will be handled in a
follow on change.

Change-Id: I6470abd7a544adf21ad143fda5da57cddad126e1
---
 .../main/java/com/google/gerrit/client/gerrit.css  |    3 +++
 .../com/google/gerrit/client/ui/LinkMenuItem.java  |    5 +++++
 2 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/gerrit.css b/gerrit-gwtui/src/main/java/com/google/gerrit/client/gerrit.css
index 5fb56ea..e85861a 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/gerrit.css
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/gerrit.css
@@ -129,6 +129,9 @@ a:hover {
   background: selectionColor;
 }
 
+.menuItem.activeRow {
+  background: selectionColor;
+}
 
 /** CommentPanel **/
 .commentPanelBorder {
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/ui/LinkMenuItem.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/ui/LinkMenuItem.java
index f61d79c..20569c3 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/ui/LinkMenuItem.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/ui/LinkMenuItem.java
@@ -33,5 +33,10 @@ public class LinkMenuItem extends InlineHyperlink implements ScreenLoadHandler {
   }
 
   public void onScreenLoad(ScreenLoadEvent event) {
+    if (event.getScreen().getToken().equals(getTargetHistoryToken())){
+      addStyleName(Gerrit.RESOURCES.css().activeRow());
+    } else {
+      removeStyleName(Gerrit.RESOURCES.css().activeRow());
+    }
   }
 }
-- 
1.7.8

